### ISSUE TITLE
System.ArgumentOutOfRangeException when using JsonConvert.DeserializeXmlNode in .NET Core

### ISSUE THREAD

**Issue by faarbaek** [JamesNK/Newtonsoft.Json#1439]
### Expected behavior
When calling JsonConvert.DeserializeXmlNode in a .NET Core 2.0 project with JSON containing a time field with the value of DateTime.MinValue, it should return the corresponding XML without throwing any exceptions. 

### Actual behavior

JsonConvert.DeserializeXmlNode throws the following exception

```csharp
System.ArgumentOutOfRangeException occurred
  HResult=0x80131502
  Message=The UTC time represented when the offset is applied must be between year 0 and 10,000.
  Source=<Cannot evaluate the exception source>
  StackTrace:
   at System.DateTimeOffset.ValidateDate(DateTime dateTime, TimeSpan offset)
   at System.DateTimeOffset..ctor(DateTime dateTime)
   at Newtonsoft.Json.Converters.XmlNodeConverter.ConvertTokenToXmlValue(JsonReader reader)
   at Newtonsoft.Json.Converters.XmlNodeConverter.CreateElement(JsonReader reader, IXmlDocument document, IXmlNode currentNode, String elementName, XmlNamespaceManager manager, String elementPrefix, Dictionary`2 attributeNameValues)
   at Newtonsoft.Json.Converters.XmlNodeConverter.ReadElement(JsonReader reader, IXmlDocument document, IXmlNode currentNode, String propertyName, XmlNamespaceManager manager)
   at Newtonsoft.Json.Converters.XmlNodeConverter.DeserializeNode(JsonReader reader, IXmlDocument document, XmlNamespaceManager manager, IXmlNode currentNode)
   at Newtonsoft.Json.Converters.XmlNodeConverter.CreateElement(JsonReader reader, IXmlDocument document, IXmlNode currentNode, String elementName, XmlNamespaceManager manager, String elementPrefix, Dictionary`2 attributeNameValues)
   at Newtonsoft.Json.Converters.XmlNodeConverter.ReadElement(JsonReader reader, IXmlDocument document, IXmlNode currentNode, String propertyName, XmlNamespaceManager manager)
   at Newtonsoft.Json.Converters.XmlNodeConverter.DeserializeNode(JsonReader reader, IXmlDocument document, XmlNamespaceManager manager, IXmlNode currentNode)
   at Newtonsoft.Json.Converters.XmlNodeConverter.CreateElement(JsonReader reader, IXmlDocument document, IXmlNode currentNode, String elementName, XmlNamespaceManager manager, String elementPrefix, Dictionary`2 attributeNameValues)
   at Newtonsoft.Json.Converters.XmlNodeConverter.ReadElement(JsonReader reader, IXmlDocument document, IXmlNode currentNode, String propertyName, XmlNamespaceManager manager)
   at Newtonsoft.Json.Converters.XmlNodeConverter.ReadJson(JsonReader reader, Type objectType, Object existingValue, JsonSerializer serializer)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.DeserializeConvertable(JsonConverter converter, JsonReader reader, Type objectType, Object existingValue)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonConvert.DeserializeObject(String value, Type type, JsonSerializerSettings settings)
   at Newtonsoft.Json.JsonConvert.DeserializeXmlNode(String value, String deserializeRootElementName, Boolean writeArrayAttribute)
```
NOTE: The exception does not occur when the time zone is GMT (UTC), but occurs when timezone is e.g. CET (UTC+1)
Works fine in Framework 4.6.1

### Steps to reproduce
In a .NET Core project (e.g. 2.0) execute the following code 
```csharp
JsonConvert.DeserializeXmlNode("{Time: \"0001-01-01T00:00:00\"}")
```

This is due to the fact, that internally it calls XmlConvert.ToString(DateTimeOffset, string), as XmlConvert.ToString(DateTime, string) is not available in .NET Standard 1.3

**Comment by moh-hassan** [comment]
This is expected because you install-package from Package Console Manager and the nearest compatible version is netstandard1.3 which is installed by nuget manager.

You can use the  TFM net45 version (the full featured) instead of netstandard1.3.

**Steps:**
- Create .NetStandard 2 class library project. 

- Download manually json.net: https://www.nuget.org/api/v2/package/Newtonsoft.Json/10.0.3
- Reference the net45 version by browsing the the location of the package in disk.
- Add you code that call xml converter:

		 public  class Json45
			{				
				public  void Test1()
				{
					var xml = JsonConvert.DeserializeXmlNode("{Time: \"0001-01-01T00:00:00\"}");
					Console.WriteLine(xml.InnerXml);					
				}
			} 

- In netcore2 project reference your netstandard2 project
- call your methods

>result
> `<Time>0001-01-01T00:00:00</Time>`

Note: if you try to reference json.net net45 from .netcore2 project, you get an error "invalid or not supported"

I implemented that solution in vs 2017.3 and it's working fine.
I test the portability of json.net to .netstandard2 (.netcore2)  using [.NET Portability Analyzer](https://marketplace.visualstudio.com/items?itemName=ConnieYau.NETPortabilityAnalyzer)  and it's 99%+ with very minor issues.

**Is this official solution**
yes,

>.NET Core 2.0 is able to freely reference libraries that have been built for .NET Framework up to version 4.6.1
> However, some libraries may fail at run time if they try to use API methods that aren't available on .NET Core

Reference: [.net core app target .net framework 4.5.2 on Linux](https://stackoverflow.com/a/43876022/3142139)

**Comment by JamesNK** [comment]
Fixed https://github.com/JamesNK/Newtonsoft.Json/commit/92d170a0783bc962f52faf02acfcecbe21d3b683
